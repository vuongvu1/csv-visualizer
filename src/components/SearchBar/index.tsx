import React, { useState, useEffect, useCallback } from "react";
import debounce from "utils/debounce";
import SC from "./styles";

interface Props {
  onSearch?: (text: string) => void;
}

const SearchBar: React.FC<Props> = ({ onSearch }) => {
  const [text, setText] = useState("");

  const searchWithDebounce = useCallback(
    debounce((keywords: string) => {
      onSearch?.(keywords);
    }, 300),
    []
  );

  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
    // if press [Enter] key
    if (event.keyCode === 13 && text) {
      searchWithDebounce(text);
    }
  };

  useEffect(() => {
    searchWithDebounce(text);
  }, [searchWithDebounce, text]);

  return (
    <SC.Container>
      <SC.SearchInput
        data-testid="searchInputId"
        placeholder="Search..."
        onChange={(e) => setText(e.target.value)}
        onKeyDown={handleKeyDown}
        value={text}
      />
      <SC.Search
        onClick={() => searchWithDebounce(text)}
        data-testid="searchIconId"
      />
    </SC.Container>
  );
};

export default SearchBar;
