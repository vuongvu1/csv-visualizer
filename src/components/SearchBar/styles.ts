import styled from "styled-components";
import { Search } from "assets/icons";
import { neutralDarkColor } from "configs/colors";
import { FontSize } from "configs/styles";

export default {
  Container: styled.div`
    position: relative;
    width: 60%;
    height: 32px;
  `,
  SearchInput: styled.input`
    border: 2px solid ${neutralDarkColor};
    width: 100%;
    height: 100%;
    border-radius: 24px;
    font-size: ${FontSize.medium};
    padding: 0 34px 0 12px;
  `,
  Search: styled(Search)`
    position: absolute;
    top: 8px;
    right: 12px;
    cursor: pointer;
  `,
};
