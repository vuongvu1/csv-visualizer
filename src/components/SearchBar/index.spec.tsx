import React from "react";
import { render, cleanup } from "@testing-library/react";
import SearchBar from "./index";

describe("SearchBar", () => {
  afterEach(cleanup);

  test("should match snapshot", () => {
    const { container } = render(<SearchBar />);
    expect(container).toMatchSnapshot();
  });
});
