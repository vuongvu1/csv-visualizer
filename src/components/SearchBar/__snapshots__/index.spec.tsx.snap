// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should match snapshot 1`] = `
.c0 {
  position: relative;
  width: 60%;
  height: 32px;
}

.c1 {
  border: 2px solid #707070;
  width: 100%;
  height: 100%;
  border-radius: 24px;
  font-size: 16px;
  padding: 0 34px 0 12px;
}

.c2 {
  position: absolute;
  top: 8px;
  right: 12px;
  cursor: pointer;
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      placeholder="Search..."
      value=""
    />
    <svg
      class="c2"
      height="1em"
      viewBox="0 0 512.005 512.005"
      width="1em"
    >
      <path
        d="M505.749 475.587l-145.6-145.6c28.203-34.837 45.184-79.104 45.184-127.317C405.333 90.926 314.41.003 202.666.003S0 90.925 0 202.669s90.923 202.667 202.667 202.667c48.213 0 92.48-16.981 127.317-45.184l145.6 145.6c4.16 4.16 9.621 6.251 15.083 6.251s10.923-2.091 15.083-6.251c8.341-8.341 8.341-21.824-.001-30.165zM202.667 362.669c-88.235 0-160-71.765-160-160s71.765-160 160-160 160 71.765 160 160-71.766 160-160 160z"
      />
    </svg>
  </div>
</div>
`;
